packages/helpers/src/input-generators.ts

```diff
@@ -35,29 +35,126 @@ type DKIMVerificationArgs = {
   fallbackToZKEmailDNSArchive?: boolean;
 };
 
-function removeSoftLineBreaks(body: string[]): string[] {
+/**
+ * Finds a selector string in cleaned content and maps it back to its original position.
+ *
+ * @param cleanContent - Uint8Array of content with soft line breaks removed
+ * @param selector - String to find in the cleaned content
+ * @param positionMap - Map of cleaned content indices to original content indices
+ * @returns Object containing the selector and its position in original content
+ * @throws Error if selector not found or position mapping fails
+ *
+ * @example
+ * cleanContent: "HelloWorld"
+ * selector: "World"
+ * positionMap: { 5->8 } (due to removed "=\r\n")
+ * returns: { selector: "World", originalIndex: 8 }
+ */
+function findSelectorInCleanContent(
+  cleanContent: Uint8Array,
+  selector: string,
+  positionMap: Map<number, number>,
+): { selector: string; originalIndex: number } {
+  const cleanString = new TextDecoder().decode(cleanContent);
+  const selectorIndex = cleanString.indexOf(selector);
```

Comment:
Do these lines work even if `cleanContent` contains multi-bytes characters in UTF8?